sub-packages/bionemo-geometric/src/bionemo/geometric/atom_featurizers.py

```diff
@@ -0,0 +1,398 @@
+
+
+class SmartsFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by hydrogen donor/acceptor and acidity/basicity."""
+
+    def __init__(self):
+        """Initializes SmartsFeaturizer class."""
+        self.hydrogen_donor = Chem.MolFromSmarts("[$([N;!H0;v3,v4&+1]),$([O,S;H1;+0]),n&H1&+0]")
+        self.hydrogen_acceptor = Chem.MolFromSmarts(
+            "[$([O,S;H1;v2;!$(*-*=[O,N,P,S])]),$([O,S;H0;v2]),$([O,S;-]),$([N;v3;!$(N-*=[O,N,P,S])]),"
+            "n&H0&+0,$([o,s;+0;!$([o,s]:n);!$([o,s]:c:n)])]"
+        )
+        self.acidic = Chem.MolFromSmarts("[$([C,S](=[O,S,P])-[O;H1,-1])]")
+        self.basic = Chem.MolFromSmarts(
+            "[#7;+,$([N;H2&+0][$([C,a]);!$([C,a](=O))]),$([N;H1&+0]([$([C,a]);!$([C,a](=O))])[$([C,a]);"
+            "!$([C,a](=O))]),$([N;H0&+0]([C;!$(C(=O))])([C;!$(C(=O))])[C;!$(C(=O))])]"
+        )
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return 4
+
+    def compute_features(self, mol):
+        """Computes matches by prefixed SMARTS patterns."""
+        hydrogen_donor_match = sum(mol.GetSubstructMatches(self.hydrogen_donor), ())
+        hydrogen_acceptor_match = sum(mol.GetSubstructMatches(self.hydrogen_acceptor), ())
+        acidic_match = sum(mol.GetSubstructMatches(self.acidic), ())
+        basic_match = sum(mol.GetSubstructMatches(self.basic), ())
+
+        # Set everything to False by default
+        for atom in mol.GetAtoms():
+            atom.SetBoolProp("hba", False)
+            atom.SetBoolProp("hbd", False)
+            atom.SetBoolProp("acidic", False)
+            atom.SetBoolProp("basic", False)
+
+        # Set matching substructures to True
+        for idx in hydrogen_acceptor_match:
+            mol.GetAtomWithIdx(idx).SetBoolProp("hba", True)
+
+        for idx in hydrogen_donor_match:
+            mol.GetAtomWithIdx(idx).SetBoolProp("hbd", True)
+
+        for idx in acidic_match:
+            mol.GetAtomWithIdx(idx).SetBoolProp("acidic", True)
+
+        for idx in basic_match:
+            mol.GetAtomWithIdx(idx).SetBoolProp("basic", True)
+
+        return mol
+
+    def get_features(self, atom):
+        """Returns features of the atom."""
+        return get_boolean_atomic_prop(atom, prop_list=["hba", "hbd", "acidic", "basic"])
+
+
+class CrippenFeaturizer(BaseFeaturizer):
+    """Class for featurizing atom by Crippen logP and molar refractivity."""
+
+    def __init__(self):
+        """Initializes CrippenFeaturizer class."""
+        pass
+
+    @property
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return 2
+
+    def compute_features(self, mol):
+        """Compute atomic contributions to Crippen logP and molar refractivity."""
+        logp_mr_list = np.array(rdMolDescriptors._CalcCrippenContribs(mol))
+        logp_mr_list[:, 0] = np.clip(logp_mr_list[:, 0], a_min=MIN_LOGP, a_max=MAX_LOGP)
```

Comment:
This MIN_LOGP and MAX_LOGP are only used for this one right? if so, maybe avoid define them globally

Reply:
Makes sense. I will move them into the class.